src/main/java/org/springframework/data/elasticsearch/client/elc/RequestConverter.java

```diff
@@ -1284,6 +1293,35 @@ public MsearchRequest searchMsearchRequest(
 		});
 	}
 
+	/**
+	 * {@link MsearchRequest} and {@link MsearchTemplateRequest} share the same {@link MultisearchHeader}
+	 */
+	private Function<MultisearchHeader.Builder, ObjectBuilder<MultisearchHeader>> msearchHeaderSetter(Query query,
```

Comment:
Wouldn't `msearchHeaderBuilder` be a better name for the method?

Reply:
🤔 sounds resonable